### ISSUE TITLE
Testimonials: Featured Image disappears

### ISSUE THREAD

**Issue by alaczek** [Automattic/jetpack#2491]
After making changes to either testimonials title or description, Testimonials Page Featured Image disappears.

Steps to reproduce:
1. Add Testimonials Featured Image, save. 
2. Go to testimonials archive page - the image should be there.
3. Modify testimonials page title or description in Customizer (don't touch the image), save.
4. Go to testimonials archive page and see if the image is there (for me it isn't).

Tested both on a local install and WordPress.com with Sela and Motif themes.

**Comment by jeherve** [comment]
When you attach a new image to the setting through the customizer, we use the `attachment_url_to_postid` ([ref](https://developer.wordpress.org/reference/functions/attachment_url_to_postid/)) to convert the attachment URL we get into a post ID. That's needed, since Sela uses `wp_get_attachment_image` to display the image ([ref](https://themes.trac.wordpress.org/browser/sela/1.0.5/archive-jetpack-testimonial.php#L14)), and `wp_get_attachment_image` expects a Post ID, not a URL.

However, when you go back and save settings again, the Featured Image setting is overwritten, and the option now uses an image URL instead of a Post ID.

This seems to be caused by this: https://github.com/Automattic/jetpack/blob/3.6.1/modules/custom-post-types/testimonial.php#L482

This code was added by 4ee37f63099a60fe00916d73bb615052a685140f. The change was made on WordPress.com first, by @mattwiebe to fix #7251-wpcom.
I tried reproducing the bug on a Jetpack site and on a WordPress.com site by removing the `coerce_testimonial_image_to_url` function, but everything seemed to work properly. 
@mattwiebe Is it possible that this workaround is no longer necessary? Would it be safe to remove that function?
